modules/polaris/src/main/ts/ephox/polaris/words/Words.ts

```diff
@@ -38,25 +39,57 @@ export interface WordsWithIndices<T> {
   readonly indices: WordIndex[];
 }
 
+// const isAnAcronym = (str: string) => new RegExp(/^([A-Z]\.)+$/).test(str);
+
+interface CouldBeAnAcronym {
+  prev: undefined | number | string;
+  check: undefined | boolean;
+  isTheLast: boolean;
+}
+
 const findWordsWithIndices = <T>(chars: Word<T>, sChars: string[], characterMap: CharacterMap, options: WordOptions): WordsWithIndices<T> => {
   const words: Word<T>[] = [];
   const indices: WordIndex[] = [];
   let word: Word<T> = [];
 
+  const couldBeAnAcronym: CouldBeAnAcronym = {
+    prev: undefined,
+    check: undefined,
+    isTheLast: false
+  };
+
   // Loop through each character in the classification map and determine whether
   // it precedes a word boundary, building an array of distinct words as we go.
   for (let i = 0; i < characterMap.length; ++i) {
 
     // Append this character to the current word.
     word.push(chars[i]);
 
+    const onWordBoundary = isWordBoundary(characterMap, i);
+    const dotAfterLetter = sChars[i] === '.' && couldBeAnAcronym.prev === ci.ALETTER;
+    const letterAfterDot = characterMap[i] === ci.ALETTER && (couldBeAnAcronym.prev === '.' || couldBeAnAcronym.prev === undefined);
```

Comment:
This should let you avoid keeping track of `prev` at all? Or am I misreading the algorithm? It's really confusing how you've put both characters and mapped characters into a single `undefined | number | string` variable.
```suggestion
    const dotAfterLetter = sChars[i] === '.' && characterMap[i - 1] === ci.ALETTER;
    const letterAfterDot = characterMap[i] === ci.ALETTER && (sChars[i - 1] === '.');
```

Reply:
Yeah this became hard to read.

It needs to check more than just the previous character so we can't use a simple - 1 index. Since this could be things like `U.S.A.` but we could do a loop backwards but that would mean that it could be slow if we have a lot of these patterns. Maybe we limit how many steps backwards it can take. Also looping would break this early for the happy path most words will not be single letters separated with dots. So checking every word agains a regexp should be slower for normal documents since this will likely never backtrack more than two letters.

I wonder if this check could then be moved to the isWordBoundary function it would need to get the chars array passed in as well to check for `.` characters since, we don't have that explicit in the map, it could then be a extra check at the end that checks for patterns backwards. Loops from the current index and checks for ALETTER and dots. Then you make a similar comment as the ones that are already there for edge cases like braking in numbers etc since this is similar to that.

Reply:
Thinking on this some more the only time this will trigger is if the word boundary character is a `.` so the happy path is even happier it would only execute at the end of sentences or on acronyms. So you could just make a if statement to check if the current char is `.` then then check I guess forward and backwards for single letters interspersed with dots.

Reply:
I simplified it, I try to refactor it more to see if I can make it more readable, let me know what do you think about the current version

Reply:
ok, thanks to @spocke we should have a final version